### PR TITLE
evm_transfer cli: wait for and output TxAttempt

### PR DESCRIPTION
**Pull request by aalu1418**
ticket: https://smartcontract-it.atlassian.net/browse/BCI-1805

previously: the CLI out put for transferring EVM funds only returns the sender + recipient
now: the CLI will retrieve the tx with a tx attempt which also includes the txhash, nonce, and status

### CODE REVIEW

**Review by jmank88 — DISMISSED** [review]

**Review by dimriou — APPROVED** [review]

File: core/web/evm_transfer_controller_test.go [hunk 1]
```diff
@@ -279,21 +294,90 @@ func TestTransfersController_CreateSuccess_eip1559(t *testing.T) {
...
 		Amount:             amount,
+		WaitAttemptTimeout: &timeout,
 	}
 
 	body, err := json.Marshal(&request)
 	assert.NoError(t, err)
 
 	resp, cleanup := client.Post("/v2/transfers", bytes.NewBuffer(body))
-	t.Cleanup(cleanup)
+	defer t.Cleanup(cleanup)
```
Comment (jmank88) [inline core/web/evm_transfer_controller_test.go:309]: Why defer here? Should be either-or, not both, no?
```suggestion
	t.Cleanup(cleanup)
```

File: core/web/evm_transfer_controller.go [hunk 2]
```diff
@@ -107,3 +128,32 @@ func ValidateEthBalanceForTransfer(c *gin.Context, chain evm.Chain, fromAddr com
...
+				return attempt, fmt.Errorf("failed to find transaction: %w", err)
+			}
+		}
+
+		// exit if tx attempts are found
+		if len(etx.TxAttempts) > 0 {
+			break
+		}
+		tick = time.After(recheckTime)
+	}
```
Comment (jmank88) [inline core/web/evm_transfer_controller.go:153]: Could save some lines if this format is readable :shrug: 
```suggestion
	ctx, cancel := context.WithTimeout(ctx, timeout)
	defer cancel()
	// until attempts are found or errors encountered
	for tick := time.After(0); len(etx.TxAttempts) == 0; tick = time.After(recheckTime) {
		select {
		case <-ctx.Done():
			return attempt, fmt.Errorf("%w - tx may still have been broadcast", ctx.Err())
		case <-tick:
			etx, err = FindTxWithAttempts(etx.ID)
			if err != nil {
				return attempt, fmt.Errorf("failed to find transaction: %w", err)
			}
		}
	}
```

File: core/web/evm_transfer_controller_test.go (hunk 1 above)
Comment (aalu1418) [reply core/web/evm_transfer_controller_test.go:309]: ah good catch - that's leftover from trying to figure out why the db triggers weren't working. will remove

File: core/web/evm_transfer_controller.go (hunk 2 above)
Comment (aalu1418) [reply core/web/evm_transfer_controller.go:153]: it's definitely more condensed - but it feels a little harder to understand. you really have to think about the different parts of the `for` loop to understand what it's checking for compared to a very obvious exit check?

**Review by jmank88 — APPROVED** [review]

**Review by prashantkumar1982 — APPROVED** [review]
So this logic doesn't guarantee that the txAttempt we return will actually make it onchain.
But I guess that is ok. The goal isn't to show the finalized Tx.

**Pull request merged**